pandas/core/groupby/groupby.py

```diff
@@ -2298,6 +2299,13 @@ def nth(
         1  1  2.0
         4  2  5.0
         """
+        from collections.abc import Iterable
+        if isinstance(n, Iterable):
+            return self.rows[tuple(n)]
+
+        elif isinstance(n, slice):
+            return self.rows[n]
+
```

Comment:
can you pass the not dropna case as well to .rows (e.g. L2313)?

Reply:
Done